File: src/main/java/ninja/hudy/infosmog/repository/CityRepository.java
package ninja.hudy.infosmog.repository;

import ninja.hudy.infosmog.model.CityRegion;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.Set;

@Repository
public interface CityRepository extends JpaRepository<CityRegion, Long> {
    Set<CityRegion> findByName(String name);

    Optional<CityRegion> findByCode(Integer code);
}

File: src/main/kotlin/pl/org/smok/infosmog/repository/CityRepository.kt
package pl.org.smok.infosmog.repository

import pl.org.smok.infosmog.model.CityRegion
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

import java.util.Optional

@Repository
interface CityRepository : JpaRepository<CityRegion, Long> {
    fun findByName(name: String): Set<CityRegion>

    fun findByCode(code: Int?): Optional<CityRegion>
}
